### ISSUE TITLE
Undefined WithBasePath function, and could not import the openai package

### ISSUE THREAD

**Issue by Arundas666** [amikos-tech/chroma-go#171]
```go
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	chroma "github.com/amikos-tech/chroma-go"
	"github.com/amikos-tech/chroma-go/collection"
	openai "github.com/amikos-tech/chroma-go/pkg/embeddings/openai"
	"github.com/amikos-tech/chroma-go/types"
)

func main() {
	// Create new OpenAI embedding function

	openaiEf, err := openai.NewOpenAIEmbeddingFunction(os.Getenv("OPENAI_API_KEY"))
	if err != nil {
		log.Fatalf("Error creating OpenAI embedding function: %s \n", err)
	}
	// Create a new Chroma client
	client,err := chroma.NewClient(chroma.WithBasePath("http://localhost:8000"))
	if err != nil {
        log.Fatalf("Error creating client: %s \n", err)
    }

	// Create a new collection with options
	newCollection, err := client.NewCollection(
		context.TODO(),
        "test-collection",
		collection.WithMetadata("key1", "value1"),
		collection.WithEmbeddingFunction(openaiEf),
		collection.WithHNSWDistanceFunction(types.L2),
	)
	if err != nil {
		log.Fatalf("Error creating collection: %s \n", err)
	}

	// Create a new record set with to hold the records to insert
	rs, err := types.NewRecordSet(
		types.WithEmbeddingFunction(openaiEf),
		types.WithIDGenerator(types.NewULIDGenerator()),
	)
	if err != nil {
		log.Fatalf("Error creating record set: %s \n", err)
	}
	// Add a few records to the record set
	rs.WithRecord(types.WithDocument("My name is John. And I have two dogs."), types.WithMetadata("key1", "value1"))
	rs.WithRecord(types.WithDocument("My name is Jane. I am a data scientist."), types.WithMetadata("key2", "value2"))

	// Build and validate the record set (this will create embeddings if not already present)
	_, err = rs.BuildAndValidate(context.TODO())
	if err != nil {
		log.Fatalf("Error validating record set: %s \n", err)
	}

	// Add the records to the collection
	_, err = newCollection.AddRecords(context.Background(), rs)
	if err != nil {
		log.Fatalf("Error adding documents: %s \n", err)
	}

	// Count the number of documents in the collection
	countDocs, qrerr := newCollection.Count(context.TODO())
	if qrerr != nil {
		log.Fatalf("Error counting documents: %s \n", qrerr)
	}

	// Query the collection
	fmt.Printf("countDocs: %v\n", countDocs) //this should result in 2
	qr, qrerr := newCollection.Query(context.TODO(), []string{"I love dogs"}, 5, nil, nil, nil)
	if qrerr != nil {
		log.Fatalf("Error querying documents: %s \n", qrerr)
	}
	fmt.Printf("qr: %v\n", qr.Documents[0][0]) //this should result in the document about dogs
}
```

In the above code, when I try to implement this in a sample go program, It is showing that the function WithBasePath  is undefined, and the package openai also not-found. I installed the version v0.1.4
<img width="1433" alt="Screenshot 2024-10-13 at 11 55 28 PM" src="https://github.com/user-attachments/assets/2a987a7f-55e8-48df-9992-61bfdd439eab">

**Comment by Arundas666** [comment]
Also, if we are trying to import the open-ai package like these, it is working
`
openai "github.com/amikos-tech/chroma-go/openai"
`

**Comment by tazarov** [comment]
hey @Arundas666, thanks for reporting this. I will look into this with priority.

**Comment by tazarov** [comment]
@Arundas666 found the problem. The changes you are trying to use, and I foolishly document, hoping to release them in a tagged version soon, are, in fact, not released under a tagged version. If you use:

```bash
go get github.com/amikos-tech/chroma-go@main
```

I've tested with both: `go get github.com/amikos-tech/chroma-go` (which pulls `v0.1.4`) and the above, and I faced the same issue as you with `v0.1.4`.

I'm working to release `v0.2.0` soon
